sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -734,4 +623,20 @@ private RecognizeOptions getRecognizeOptionsProperties(RecognizeOptions userProv
             return new RecognizeOptions();
         }
     }
+
+    /**
+     * Helper method that throws a {@link FormRecognizerException} if {@link AnalyzeOperationResult#getStatus()} is
+     * {@link OperationStatus#FAILED}.
+     *
+     * @param analyzeResponse The response returned from the service.
+     */
+    private void throwIfAnalyzeStatusInvalid(AnalyzeOperationResult analyzeResponse) {
+        if (OperationStatus.FAILED == analyzeResponse.getStatus()) {
+            List<ErrorInformation> errorInformationList = analyzeResponse.getAnalyzeResult().getErrors();
+            if (!CoreUtils.isNullOrEmpty(errorInformationList)) {
```

Comment:
What happens if the errorInformationList is null or empty? The status has failed but this method won't throw any exception which might result in the map operation above to continue with the transformation instead of throwing.

Reply:
Good point. We are trying to remove the method `throwIfAnalyzeStatusInvalid()` here and directly throw error when status is failed: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java#L720

Reply:
>What happens if the errorInformationList is null or empty? The status has failed but this method won't throw any exception which might result in the map operation above to continue with the transformation instead of throwing.

@srnagar 
I will cover this in https://github.com/Azure/azure-sdk-for-java/issues/12200#issuecomment-659011304.

Reply:
Removed this in - https://github.com/Azure/azure-sdk-for-java/pull/13247/files#diff-f3e13c2bd95cecb8a90329a3c1e1f6dbL632